### DISCUSSION TITLE
Mount USB disk while WSL Ubuntu boots. Is it possible?

### DISCUSSION THREAD

**Discussion by jactra1** [dorssel/usbipd-win#869] [Q&A]
Hello,

I installed Ubuntu 22.04 on WSL. I manually installed Nextcloud 28.0.2 on it. I would like to keep my data on extrernal USB Disk. The command "lsblk" don't show my disk. So, to mount USB Disk in Ubuntu on WSL I had to compile my kernel with USB Mass Storage driver. Now I mount my usb Disk in 3 steps:

1. Run WSL
2. In windows commandline run „usbipd attach —wsl —busid 3-2"
3. Back to wsl and mount /dev/sdd3 /mnt/nextcoud/data (now lsblk shows my disk and partitions including may ext4 partition on sdd3)

I installed Nextclod 28.0.2. During install process i choose to keep data on USB Disk (/mnt/nextcoud/data). Instalation is going well.
With first run, I can create admin account, login, manage data, upload photos  (photos are saved in user folder on USB Disk). etc but after restart WSL, I have to mount USB Disk again. When I do it, Nextcloud gives me an error „can’t create data folder, can’t find ,ocdata”. In addition I will say that directory on USB disk in own recursively by www-data nedded by Nextcloud. Maybe Nextcloud expects to mount partirion  on boot Ubuntu? But how to do it if I must start wsl first to run command "usbipd attach —wsl —busid 3-2"?

Please help

**Comment by dorssel** [comment]
Either restart the service after `usbipd attach` with something like `systemctl restart nextcloud`. Or make a systemd service file that calls `usbip attach` (note: without 'd') from within WSL and make sure that it is started before the nextcloud service.
